specs/networking/libp2p-standardization.md

```diff
@@ -135,24 +168,395 @@ number of shard subnets is defined via `SHARD_SUBNET_COUNT` and the shard
 Each Gossipsub
 [Message](https://github.com/libp2p/go-libp2p-pubsub/blob/master/pb/rpc.proto#L17-L24)
 has a maximum size of 512KB (estimated from expected largest uncompressed block
-size).
+size). Clients SHOULD reject and MUST NOT send messages that are over this size limit.
 
-The `data` field of a Gossipsub `Message` is an SSZ-encoded object. For the `beacon_block` topic,
+The `data` field of a Gossipsub `Message` is the encoded object being sent
+between peers. For the `beacon_block` topic,
 this is a `beacon_block`. For the `beacon_attestation` topic, this is
 an `attestation`.
 
-## Eth-2 RPC
+#### Encodings
 
-#### Protocol Id: `/eth/serenity/beacon/rpc/1`
+Topics are post-fixed with an encoding. Encodings define how the `data` field
+of a Gossipsub `Message` is encoded. Implementations SHOULD support the
+following encodings:
+
+- `ssz` -  This is a temporary encoding used only for initial interop and
+	testing. All objects are SSZ-encoded. Example: The beacon block topic
+	string is:
+	`beacon_block/ssz` and the `data` field of a Gossipsub `Message` is an
+	ssz-encoded `BeaconBlock`.
+- `ssz_snappy` - All objects are ssz-encoded and then compressed with `snappy`.
```

Comment:
Wouldn't it be more efficient to compress the stream *before* multiplexer (i.e. compress the whole transport stream)? May be libp2p has a suitable compressor out-of-the-box ?
@raulk ?